packages/commonwealth/server/util/cosmosCache.ts

```diff
@@ -0,0 +1,114 @@
+import type { Request, RequestHandler } from 'express';
+import { cacheDecorator } from 'common-common/src/cacheDecorator';
+
+const oneBlock = 6; // typical Cosmos block time is ~6 seconds
+
+export const cosmosCacheRPC = (duration: number): RequestHandler => {
+  // RPC specific codes from cosmJS requests
+  const activeCodes = ['0801', '0802']; // ['DepositPerion', 'VotingPeriod']
+  const completedCodes = [
+    '0803', // 'Passed'
+    '0804', // 'Rejected'
+    '0805', // 'Failed'
+    '0803220a0a080000000000000087', // 'Passed'
+    '0803220a0a080000000000000100', // 'Passed'
+    '0803220a0a080000000000000189', // 'Passed'
+  ];
+
+  return async function cacheIfCacheworthyRPC(req, res, next) {
```

Comment:
- lets discuss if its right place for this function
- key generator might be better file for this function

Reply:
follow up task:
https://github.com/hicommonwealth/commonwealth/issues/3665